core-api/src/main/java/com/optimizely/ab/decision/evaluator/DecisionAudienceEvaluator.java

```diff
@@ -0,0 +1,108 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+package com.optimizely.ab.decision.evaluator;
+
+import com.optimizely.ab.config.Experiment;
+import com.optimizely.ab.config.ProjectConfig;
+import com.optimizely.ab.config.audience.AudienceIdCondition;
+import com.optimizely.ab.config.audience.Condition;
+import com.optimizely.ab.config.audience.OrCondition;
+import com.optimizely.ab.event.internal.UserContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Determines whether a user satisfies audience conditions for the experiment.
+ */
+public class DecisionAudienceEvaluator implements AudienceEvaluator {
+
+    private static final Logger logger = LoggerFactory.getLogger(DecisionAudienceEvaluator.class);
+
+       /**
+     * Validates user satisfies audience conditions for the experiment
+     *
+     * @param experiment  The Experiment the user will be bucketed into.
+     * @param userContext It have user id, attributes and a reference to the current {@link ProjectConfig}
+     * @return flag if user satisfies audience conditions for the experiment
+     */
+    @Override
+    public boolean evaluate(@Nonnull Experiment experiment,
+                            @Nonnull UserContext userContext) {
+        if (experiment.getAudienceConditions() != null) {
+            return evaluateAudienceConditions(experiment, userContext);
+        } else {
+            Boolean resolveReturn = evaluateAudience(experiment, userContext);
+            return Boolean.TRUE.equals(resolveReturn);
+        }
+    }
+
+    /**
+     * Evaluate audience if experiment has no conditions
+     *
+     * @param experiment  The Experiment the user will be bucketed into.
+     * @param userContext It have user id, attributes and a reference to the current {@link ProjectConfig}
+     * @return flag if audience are valid for experiment
+     */
+    @Nullable
+    private static Boolean evaluateAudience(@Nonnull Experiment experiment,
```

Comment:
we were using annotation `visiblefortesting` just curious where is that?

Reply:
As far as I have seen, we were not using visiblefortesting annotation on this method.